redbot/cogs/streams/streams.py

```diff
@@ -279,27 +279,33 @@ async def streamalert(self, ctx: commands.Context):
         pass
 
     @streamalert.group(name="twitch", invoke_without_command=True)
-    async def _twitch(self, ctx: commands.Context, channel_name: str = None):
+    async def _twitch(
+        self, ctx: commands.Context, channel_name: str = None, discord_channel: str = None
```

Comment:
if you change all instances of `discord_channel: str` to `discord_channel: discord.TextChannel`, it becomes unnecessary to use regex to get the channel object as you will already have the channel object. I would also make sure all supported streaming services can have a discord channel specified (at the moment, Hitbox and Picarto seem to be excluded on this)

Reply:
Got it, thank you, I've made the changes you asked for. After implementing the change for Picarto and Hitbox, I saw that I couldn't use token_name as platform name because theirs are initialized as None, so I made an attribute in the Stream class to hold the platform name. Is that and my other changes all right?